openapi.yaml

```diff
@@ -12,106 +12,458 @@ info:
     email: isdataarchitecture@oregonstate.edu
 externalDocs:
   description: GitHub Repository
-  url: https://github.com/osu-mist/express-api-skeleton
+  url: https://github.com/osu-mist/library-api
 servers:
-  - url: https://api.oregonstate.edu/v2
+  - url: https://api.oregonstate.edu/v1
 security:
   - OAuth2:
       - full
 paths:
-  /pets:
+  /books:
     get:
-      summary: List all pets
+      summary: Retrieve a list of all books in the library.
       tags:
-        - pets
-      description: Get all pets
-      operationId: getPets
+        - books
+      description: Get all books
+      operationId: getBooks
       parameters:
         - $ref: '#/components/parameters/pageNumber'
         - $ref: '#/components/parameters/pageSize'
         - in: query
-          name: filter[species]
-          description: Search by pet species
+          name: filter[title]
+          description: Search by book title
           required: false
           schema:
             type: string
         - in: query
-          name: filter[hasOwner]
-          description: Filter results if pet has an owner or not
+          name: filter[author]
+          description: Search by book author.
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: filter[publicationYear]
+          description: Filter books published on a specific year.
+          required: false
+          schema:
+            type: integer
+            format: int32
+        - in: query
+          name: filter[isbn]
+          description: Filter books by International Standard Book Number (ISBN) for the book
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: filter[available]
```

Comment:
Is a book only have two statuses: available/unavailable? If we have more than one status, how would you handle them?

Reply:
this is boolean, so true or false would be the value only. Should I add more statuses I can do enum?

Reply:
Ok, sounds good to me.